### PR TITLE
Monuments index

### PR DESCRIPTION
**Pull request by Glod33**
First draft

### CODE REVIEW

**Comment by Glod33** [comment]
Screenshot 2023-03-09 at 13 03 37

**Comment by wJoenn** [comment]
I'm really not a fan of using margin and border radius on mobiles.
I think a design like Tripadvisor's would be more fitting
image

**Comment by wJoenn** [comment]
Here’s another possible inspiration. I like the layout idea but I’m not very found of the card’s appearance

**Comment by wJoenn** [comment]
Here’s another possible inspiration. I like the layout idea but I’m not very found of the card’s appearance
C9B7CF01-97CE-495E-991D-12C9B0E14636

**Comment by wJoenn** [comment]
Also I think having a distance from you like
“300m away” or something would be nice

**Comment by wJoenn** [comment]
Figured google could also be a source pf inspiration

**Comment by Glod33** [comment]
Screenshot 2023-03-09 at 18 04 16

Not too happy with it, what do y'all think?

**Comment by MarianaCarlota** [comment]
I liked the idea with the photo full width in the screen and semi-tranparent part on the bottom for the text from Louis. Maybe with blue instead of red

**Comment by Glod33** [comment]
Screenshot 2023-03-10 at 11 58 20

**Comment by Glod33** [comment]
I think it's almost ready to push. Just refactoring and removing unnecessary CSS.

I would like to add some more features, but I will do that in a separate issue.

**Comment by Glod33** [comment]
Screenshot 2023-03-10 at 18 02 41

**Comment by Glod33** [comment]
Screenshot 2023-03-10 at 18 43 03

File: app/controllers/monuments_controller.rb [hunk 1]
```diff
@@ -2,7 +2,15 @@ class MonumentsController < ApplicationController
...
-    @monuments = policy_scope(Monument).near([user.lat, user.lng], 5)
+    @user = User.find(144)
+    @monuments = policy_scope(Monument)
+
+    @markers = @monuments.geocoded.map do |monument|
+      {
+        lat: monument.lat,
+        lng: monument.lng
+      }
+    end
```
Comment (wJoenn) [inline app/controllers/monuments_controller.rb:13]: You're displaying all of the monuments now ?
Not only those around the user ?
Also, the Monuments already have a lat and a lng so you don't need to #geocoded before mapping.
You can just do `@monuments.map { |monument| { lat: monument.lat, lng: monument.lng } }`

The `User.find(144)` is only gonna work for you and not after you reseed or something. Use `.find(name: "Rutger")` instead please

File: app/javascript/controllers/map_controller.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    this.#fitMapToMarkers()
+  }
+
+  #addMarkersToMap() {
+    this.markersValue.forEach((marker) => {
+      new mapboxgl.Marker()
+        .setLngLat([ marker.lng, marker.lat ])
+        .addTo(this.map)
+    })
+  }
```
Comment (wJoenn) [inline app/javascript/controllers/map_controller.js:27]: You don't need the parentheses around the parameter in an arrow function when you only have one.
Can you remove them around `marker` in line 22 please ?

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -8,6 +8,9 @@
 
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_include_tag "application", "data-turbo-track": "reload", defer: true %>
+
+    <script src='https://api.mapbox.com/mapbox-gl-js/v2.13.0/mapbox-gl.js'></script>
+    <link href='https://api.mapbox.com/mapbox-gl-js/v2.13.0/mapbox-gl.css' rel='stylesheet' />
```
Comment (wJoenn) [inline app/views/layouts/application.html.erb:13]: Can you move these above our css please ?
That way we can overwrite it if we want to.

File: app/views/monuments/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<div class="index-container">
+  <div class="monuments-map"
+  data-controller="map"
+  data-map-markers-value="<%= @markers.to_json %>"
+  data-map-api-key-value="<%= ENV['MAPBOX_API_KEY'] %>">
```
Comment (wJoenn) [inline app/views/monuments/index.html.erb:5]: Can you send us the API key you used in DM on Slack ?
Is it the same as the previous project ?

File: app/assets/stylesheets/config/_colors.scss [hunk 5]
```diff
@@ -1,6 +1,9 @@
 // Define variables for your color scheme
 
 // For example:
+$background: #363e51;
+$navy: #21252e;
+$rose: #e7b5aa;
```
Comment (wJoenn) [inline app/assets/stylesheets/config/_colors.scss:6]: Can you move these further down after the "custom colors" comment ?
The background and rose should already be there

**Review by wJoenn — CHANGES_REQUESTED** [review]
It would have been better to have a separate branch for the map and the index instead of miwing 2 issues at once.
Please try to keep that in mind for later

File: app/assets/stylesheets/components/_index.scss [hunk 6]
```diff
@@ -1,3 +1,5 @@
+@import "components/cards";
+@import "components/containers";
```
Comment (wJoenn) [inline app/assets/stylesheets/components/_index.scss:2]: You're already inside the components, you don't need to specify it again

File: app/assets/stylesheets/config/_colors.scss [hunk 7]
```diff
@@ -12,3 +12,7 @@ $light-gray: #F4F4F4;
 $background: #363e51;
 $rose: #e7b5aa;
 $red: rgb(170, 50, 50);
+
+// custom colors
+$background: #363e51;
+$rose: #e7b5aa;
```
Comment (wJoenn) [inline app/assets/stylesheets/config/_colors.scss:18]: If I'm not mistaken you have the rose and the background two times and you don't have the navy color anymore

File: app/javascript/controllers/index.js [hunk 8]
```diff
@@ -13,6 +13,12 @@ application.register("banner-scroller", BannerScrollerController)
 import GeolocationController from "./geolocation_controller"
 application.register("geolocation", GeolocationController)
 
+import HelloController from "./hello_controller"
+application.register("hello", HelloController)
+
```
Comment (wJoenn) [inline app/javascript/controllers/index.js:18]: You can remove the HelloController, the file doesn't exist anymore

**Review by wJoenn — CHANGES_REQUESTED** [review]

**Review by wJoenn — APPROVED** [review]

**Pull request merged**